fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/api/registry/LandPathNodeTypesRegistry.java

```diff
@@ -0,0 +1,223 @@
+	 *
+	 * @param block    Block to register.
+	 * @param provider {@link DynamicPathNodeTypeProvider} to associate with the block.
+	 */
+	public static void registerDynamic(Block block, DynamicPathNodeTypeProvider provider) {
+		Objects.requireNonNull(block, "Block cannot be null!");
+		Objects.requireNonNull(provider, "DynamicPathNodeTypeProvider cannot be null!");
+
+		// Registers the provider.
+		PathNodeTypeProvider old = NODE_TYPES.put(block, provider);
+
+		if (old != null) {
+			LOGGER.debug("Replaced PathNodeType provider for the block {}", block);
+		}
+	}
+
+	/**
+	 * Gets the {@link PathNodeType} from the provider registered for the specified block state at the specified position.
+	 *
+	 * <p>If no valid {@link PathNodeType} provider is registered for the block, it returns {@code null}.
+	 * You cannot use this method to retrieve vanilla block node types.
+	 *
+	 * @param state    Current block state.
+	 * @param world    Current world.
+	 * @param pos      Current position.
+	 * @param neighbor Specifies if the block is not a directly targeted block, but a neighbor block in the path.
+	 * @return the custom {@link PathNodeType} from the provider registered for the specified block,
+	 * passing the block state, the world, and the position to the provider, or {@code null} if no valid
+	 * provider is registered for the block.
+	 */
+	@Nullable
+	public static PathNodeType getPathNodeType(BlockState state, BlockView world, BlockPos pos, boolean neighbor) {
+		Objects.requireNonNull(state, "BlockState cannot be null!");
+		Objects.requireNonNull(world, "BlockView cannot be null!");
+		Objects.requireNonNull(pos, "BlockPos cannot be null!");
+
+		// Gets the node type provider for the block.
+		PathNodeTypeProvider provider = getPathNodeTypeProvider(state.getBlock());
+
+		//If no provider exists, returns null.
+		if (provider == null) return null;
+
+		//If a provider exists, returns the node type obtained from the provider.
+		//The node type can be null too.
+		if (provider instanceof DynamicPathNodeTypeProvider) {
+			return ((DynamicPathNodeTypeProvider) provider).getPathNodeType(state, world, pos, neighbor);
+		} else {
+			return ((StaticPathNodeTypeProvider) provider).getPathNodeType(state, neighbor);
+		}
+	}
+
+	/**
+	 * Gets the {@link PathNodeTypeProvider} registered for the specified block.
+	 *
+	 * <p>If no {@link PathNodeTypeProvider} is registered for the block, it returns {@code null}.
+	 *
+	 * <p>Tip: This method is intended to be used in any cases in which you need to get
+	 * the raw provider for the block, if you need the {@link PathNodeType} for the block state instead,
+	 * you can simply use {@link #getPathNodeType}.
+	 *
+	 * @param block Current block.
+	 * @return the {@link PathNodeTypeProvider} registered for the specified block,
+	 * or {@code null} if no provider is registered for the block.
+	 */
+	@Nullable
+	public static PathNodeTypeProvider getPathNodeTypeProvider(Block block) {
```

Comment:
Looks great. Could you maybe add a warning saying that we reserve the right to add more subinterfaces, and that users of this method must take that into account?